core/objects.go

```diff
@@ -599,11 +599,21 @@ var RevocationReasons = map[RevocationCode]string{
 	5: "cessationOfOperation",
 	6: "certificateHold",
 	// 7 is unused
-	8:  "removeFromCRL", // needed?
+	8:  "removeFromCRL",
 	9:  "privilegeWithdrawn",
 	10: "aAcompromise",
 }
 
+// UserRevocationReasons contains the RevocationCodes which users are allowed
+// to use
+var UserRevocationReasons = map[RevocationCode]struct{}{
```

Comment:
Could we take this opportunity to move the revocation-related constants into a `revocation` package rather than adding to `core`?

Also, I'd be happier turning the above list into an enum and indexing this map with that enum. That way instead of documenting what each of these numbers means here in a comment, and also up above, we have a single authoritative mapping of name to number. We could add a stringer (probably manually written to save dependencies) to turn the enum into a string where needed.